internal/pkg/skuba/upgrade/node/versions.go

```diff
@@ -88,6 +89,51 @@ func (nviu NodeVersionInfoUpdate) IsFirstControlPlaneNodeToBeUpgraded(client cli
 	return isControlPlane && allControlPlanesMatchVersion && matchesClusterVersion, nil
 }
 
+// IsNodeUpgradeable returns whether a given node is upgradeable or not, taking global cluster restrictions into account.
+// If all preconditions are met for the given node, no error will be returned.
+func (nviu NodeVersionInfoUpdate) IsNodeUpgradeable(client clientset.Interface, currentClusterVersion *version.Version) error {
+	var upgradeable bool
+	isFirstControlPlaneNodeToBeUpgraded, err := nviu.IsFirstControlPlaneNodeToBeUpgraded(client)
+	if err != nil {
+		return err
+	}
+	if isFirstControlPlaneNodeToBeUpgraded {
+		upgradeable, err = kubernetes.AllWorkerNodesTolerateVersion(client, nviu.Update.APIServerVersion)
+		if err != nil {
```

Comment:
Here and the two below: couldn't this be reconstructed as:

```golang
if isFirstControlPlaneNodeToBeUpgraded {
    upgradeable, err := myFunction(...)
    if upgradeable || err != nil {
        return err
    }
    error.WriteString(...)
} else {
    // The other cases
}

return error
```

Just saying because it looks like if it can be upgraded, `err` would be nil anyways, no? (and if isn't it wouldn't matter much, then we'd need to add an extra if inside of the current one). All in all, it would allow us to get rid off redundant if-else statements at the end of this function.

Reply:
Not sure what you mean, can you please elaborate?

Reply:
Or
```
if ! upgradeable || err != nil {
      errorMessages = append(errorMessages, fmt.Sprintf("There are addon upgrades available for the current cluster version (%s) that need to be applied first", currentClusterVersion))
    }

```